src/Simulation/Simulation.jl

```diff
@@ -458,6 +456,41 @@ function _guess_optimal_number_of_threads_for_SOR(gs::NTuple{3, Integer}, max_nt
     return min(nextpow(2, max(cld(n+1, 25), 4)), max_nthreads)
 end
 
+function _get_grid_for_WP(sim::Simulation{T}, contact_id::Int) where T
```

Comment:
I guess we just wan't to call this function `Grid` and use dispatch.

`Grid(sim::Simulation)` for Electric Potential
`Grid(sim::Simulation, contact_id::Int)` for a Weighting Potential.

Reply:
Sounds reasonable